infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -235,6 +226,23 @@ public synchronized void refresh(@Nonnull TopologyDescriptor newTopology) {
         if (replicationContext.getIsLeader().get()) {
             removeStaleSessionOnLeadershipAcquire();
         }
+
+        // check if any new SINKs have been added to the topology.
+        Set<String> newRemoteSinksAdded = Sets.difference(newRemoteSinks, currentRemoteSinks);
+        if (!newRemoteSinksAdded.isEmpty()) {
+            configManager.getRegisteredSubscribers().forEach(this::createSessions);
+        } else {
+            // Since sessions are created on all the nodes in the SOURCE cluster, its possible that on leadership change,
+            // the new leader doesn't have any new sessions to create. In that case, only connect
+            connectToRemoteClusters();
+        }
```

Comment:
IMO we shouldn't do this if check here at this step, since it's possible that during the leadership change a real topology change also happened. Given that `createSessions` and `connectToRemoteClusters` are already idempotent, I think we can just do both here and let next layer figure out if it is required session creation / connection

Reply:
`createSession` also calls  `connectToRemoteClusters()`.
So even if the topology changed during leadership change, the sessions would still connect to remote.

This was a small optimization to reduce duplication of computation. But if its making the code look not-simple, I'll remove the if